include/revng/ADT/STLExtras.h

```diff
@@ -53,3 +53,54 @@ concept IsNotTupleEnd
 static_assert(HasTupleSize<std::tuple<>>);
 static_assert(!HasTupleSize<std::vector<int>>);
 static_assert(!HasTupleSize<int>);
+
+//===----------------------------------------------------------------------===//
+//     Extra additions to <iterator>
+//===----------------------------------------------------------------------===//
+
+namespace revng {
+namespace detail {
+template<typename ItTy,
+         typename FuncTy,
+         typename FuncReturnTy = decltype(
+           std::declval<FuncTy>()(*std::declval<ItTy>()))>
+class proxy_mapped_iterator_impl : public llvm::mapped_iterator<ItTy, FuncTy> {
+  struct IteratorProxy {
+    FuncReturnTy Temporary;
+
+    IteratorProxy(FuncReturnTy &&Value) : Temporary(std::move(Value)) {}
+    FuncReturnTy *const operator->() { return &Temporary; }
+    FuncReturnTy const *const operator->() const { return &Temporary; }
+  };
+
+public:
+  using llvm::mapped_iterator<ItTy, FuncTy>::mapped_iterator;
```

Comment:
Is this `using` really necessary now?

Reply:
It's a constructor pass-through, As we're mirroring `llvm::mapped_iterator`, I think it's a good idea to keep the original constructor.

Reply:
😮

Reply:
I think, I'll "resolve" this one myself after all. If anything, feel free to unresolve it, or to create a new one.

Reply:
Cool! And this also just reminded me of another place, unrelated with this PR, where I probably need something like this. Thanks!